src/stdatamodels/jwst/transforms/models.py

```diff
@@ -1989,6 +1991,182 @@ def _evaluate_transform_guess_form(model, x=0, y=0, t=0):
     raise TypeError(f"Expected a model or list of models, but got {type(model)}. ")
 
 
+class MIRIWFSSBackwardDispersion(Model):
+    """Calculate the dispersion extent of MIRI WFSS pixels."""
+
+    standard_broadcasting = False
+    _separable = False
+    fittable = False
+    linear = False
+
+    n_inputs = 4
+    n_outputs = 5
+
+    def __init__(self, orders, lmodels=None, xmodels=None, ymodels=None, name=None, meta=None):
+        """
+        Initialize the model.
+
+        Parameters
+        ----------
+        orders : list
+            The list of orders which are available to the model. For MIRIWFSS orders = 1.
+        lmodels : list
+            The list of models for the polynomial model in t (inverse dispersion)
+        xmodels : list
+            The list of models for the polynomial model in x
+        ymodels : list[tuple]
+            The list of tuple(models) for the polynomial model in y
+        name : str, optional
+            Name of the model
+        meta : dict
+            Unused
+        """
+        self._order_mapping = {int(k): v for v, k in enumerate(orders)}
+        self.xmodels = xmodels
+        self.ymodels = ymodels
+        self.lmodels = lmodels
+        meta = {"orders": orders}
+        if name is None:
+            name = "miri_wfss_backward_dispersion"
+        super(MIRIWFSSBackwardDispersion, self).__init__(name=name, meta=meta)
+        self.inputs = ("x", "y", "wavelength", "order")
+        self.outputs = ("x", "y", "x0", "y0", "order")
+
+    def evaluate(self, x, y, wavelength, order):
+        """
+        Transform from the direct image plane to the dispersed plane.
+
+        Parameters
+        ----------
+        x, y : float
+            Input x, y location
+        wavelength : float
+            Wavelength in angstroms
+        order : int
+            Input spectral order
+
+        Returns
+        -------
+        x, y : float
+            The x, y values in the dispersed plane.
+        x0 y0 : float
+            Source object x-center, y-center. Same as input x, y.
+        order : int
+            Output spectral order, same as input
+        """
+        if (wavelength < 0).any():
+            raise ValueError("Wavelength should be greater than zero")
+        try:
+            iorder = self._order_mapping[int(order.flatten()[0])]
+        except KeyError as err:
+            raise ValueError("Specified order is not available") from err
+        # t is trace normalization parameters (it was values of 0 to 1)
+        t = self.tmodels[iorder](wavelength)
+        xmodel = self.xmodels[iorder]
+        ymodel = self.ymodels[iorder]
+
+        dx = xmodel(t)
```

Comment:
I don't think this will work with the current reference file and schema for it. `self.xmodels[iorder]` is a list f two models and so it can't be evaluated like this. OTH, I'm not sure why it's a list of two models and what it is supposed to be. I would have expected either one model (for no spatial dependence) or three (for spatial dependence). Is there a description of the calibration? In any case, this model and possibly the schema and model for the reference file needs to be fixed.

Reply:
I have created a new reference file the old was incorrect. 
The xmodel is just 1D model. I was waiting to deliver a new reference file to make sure we did not need any other changes to it. Right now the xmodel has no spatial dependency. - But that might change once MIRI gets data for this mode.

Reply:
@nden How is the best way to share it ? I tried to attach it but it would not let me attach an asdf file.